code/swb_player/PlayerBase.Camera.cs

```diff
@@ -0,0 +1,120 @@
+using SWB.Shared;
+using System;
+using static Sandbox.CursorSettings;
+
+namespace SWB.Player;
+
+public partial class PlayerBase
```

Comment:
I'm not a big fan of you moving this into the player class, it was specifically moved outside to separate logic and reduce bloating. The idea is that in the future this logic could be handled with an interface to allow custom camera controllers.